app/models/task_definition.rb

```diff
@@ -15,6 +15,8 @@ class TaskDefinition < ActiveRecord::Base
   belongs_to :group_set
   belongs_to :tutorial_stream
 
+  has_one :draft_task_definition_unit, foreign_key: 'draft_task_definition_id', class_name: 'Unit', dependent: :nullify
```

Comment:
This should be the other way around - this may belong to a unit as the draft learning summary.

Reply:
From my understanding, `has_one` points to a table that contains a foreign key, whereas belongs to will be the class containing the foreign key. Since a Unit has the foreign key of `draft_task_definition_id` wouldn't this be the correct way around?
A `belongs_to` relation also doesn't accept a `:nullify` dependent, which needs to be set on the `TaskDefinition`, since we aim to nullify the FK on the Unit.

I could be understanding it wrong, let me know what you think!

Reply:
That makes sense :) - just seems backward. Will this cause any issues on destroy of a unit? May be worth checking due to the circular relationship.

Reply:
> just seems backward.

I totally agree, I struggled to wrap my head around it for that very reason, the `belong` and `has` terminology seems as if it would indicate a different relation. I'll do some further testing just incase. 

> Will this cause any issues on destroy of a unit? May be worth checking due to the circular relationship.

I'll definitely test this. When I write some unit tests as well I'll test all of these things